### PR TITLE
[DO NOT MERGE]: Add support for WindowsAzure.Storage 7.0.0

### PR DESCRIPTION
**Pull request by mikeblakeuk**
Attempting to gather dependency information for package 'WindowsAzure.Storage.7.0.0' with respect to project 'MyHdiProject', targeting '.NETFramework,Version=v4.5.2'
Attempting to resolve dependencies for package 'WindowsAzure.Storage.7.0.0' with DependencyBehavior 'Lowest'
Unable to resolve dependencies. 'WindowsAzure.Storage 7.0.0' is not compatible with 'Microsoft.Azure.Management.HDInsight.Job 2.0.2 constraint: WindowsAzure.Storage (>= 6.0.0 && < 7.0.0)'.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@mikeblakeuk__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no contribution license agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, AZPRBOT;

**Comment by mikeblakeuk** [comment]
@hovsepm

**Comment by hovsepm** [comment]
@azuresdkci add to whitelist

**Comment by hovsepm** [comment]
@shefaliv please take a look

**Comment by hovsepm** [comment]
@mikeblakeuk https://github.com/Azure/azure-sdk-for-net/pull/1985 already was reviewed and closed. The issue here is that major version change (6.x.x -> 7.x.x) supposed to be breaking and we do not want to introduce here potential bugs. 
The suggested way is to have compatibility within minor versions (6.0.x -> 6.999.999.999, or 7.0 exclusive). Please provide more details why do you need this broad range of compatibility and I'l contact the owner team. Right now your PR does not meet our code review criteria.

**Comment by mikeblakeuk** [comment]
Hi,
#1985 had a typo of "8" in the title which gave of the impression that a preview build was required (or any build).
We have upgraded our solution to use Windows Storage 7.0.0 which I assume is the latest GA library (https://www.nuget.org/packages/WindowsAzure.Storage/). For some reason, HDI Jobs will only allow 6.0-6.9) 
As you can see from the visual studio error message, 7.0.0 is not compatible, hence the change to the **Microsoft.Azure.Management.HDInsight.Job.nuspec**
Can we contact the HDInight Job owner to see if they will support both 6.x and 7.x versions?
Thanks

Also, why did this fail code review?

**Comment by mikeblakeuk** [comment]
Hi, I've updated the test dependencies to also use Storage (12a01db). Hope this helps

Done Building Project "C:\git\azure-sdk-for-net\src\TestDependencies\TestDependencies.csproj" (default targets).
Done Building Project "C:\git\azure-sdk-for-net\src\ResourceManagement\HDInsightJob\HDInsightJob.sln" (Build target(s)).
Done Building Project "C:\git\azure-sdk-for-net\build.proj" (default targets).
Build succeeded.
    0 Warning(s)
    0 Error(s)
Time Elapsed 00:00:10.35

and 
image

**Comment by shefaliv** [comment]
+ Subramanyam

Get Outlook for iOS<https://aka.ms/o0ukef>




On Tue, May 17, 2016 at 10:22 AM -0700, "Michael Blake" <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi, I've updated the test dependencies to also use Storage (12a01db<https://github.com/Azure/azure-sdk-for-net/commit/12a01db6a01386ce4e3c59d915b46073433b53a4>). Hope this helps

Done Building Project "C:\git\azure-sdk-for-net\src\TestDependencies\TestDependencies.csproj" (default targets).
Done Building Project "C:\git\azure-sdk-for-net\src\ResourceManagement\HDInsightJob\HDInsightJob.sln" (Build target(s)).
Done Building Project "C:\git\azure-sdk-for-net\build.proj" (default targets).
Build succeeded.
0 Warning(s)
0 Error(s)
Time Elapsed 00:00:10.35

and
[image]<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fcloud.githubusercontent.com%2fassets%2f1986660%2f15332102%2f31c04dce-1c5c-11e6-87f4-8248283f3d4d.png&data=01%7c01%7cshvohra%40microsoft.com%7c122101b219ec4c3b060408d37e77c882%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=oA9soC8XWvrxXi%2bssrfmeP%2bOKo3Q54KBtRAgwa1c%2fuE%3d>

-

**Comment by mikeblakeuk** [comment]
I can't access the Jenkins build server so not sure how I can help?

File: src/ResourceManagement/HDInsightJob/HDInsightJob/Microsoft.Azure.Management.HDInsight.Job.nuspec [hunk 1]
```diff
@@ -84,7 +87,7 @@ The methods “GetJobOutput”(+Async) and “GetJobErrorLogsAsync”(+Async) no
     </references>
     <dependencies>
       <dependency id="Microsoft.Azure.Common" version="[2.1.0,3.0)" />
-      <dependency id="WindowsAzure.Storage" version="[6.0.0,7.0)" />
+      <dependency id="WindowsAzure.Storage" version="[7.0.0,7.0)" />
```
Comment (hovsepm) [inline src/ResourceManagement/HDInsightJob/HDInsightJob/Microsoft.Azure.Management.HDInsight.Job.nuspec]: @mikeblakeuk you should have version range -> [7.0.0, 8..0).

**Comment by mikeblakeuk** [comment]
Why can we not support versions 5,6 and 7?

**Comment by mikeblakeuk** [comment]
What else do i need to do?

**Comment by shefaliv** [comment]
@pattipaka

**Comment by markcowl** [comment]
@sheliv @pattipaka @mikeblamnkeuk.  If you plan never to use this library in azure powershell, then you can update to storage versionn 7.  Meanwhile, this would require an across-the-board change to all dependencies on windows azure storage for all other rmodules.  Unless you have an extremely compelling reason for updating, we cannot accept this.  Even with an extremely compelling reason, it will take coordination with other teams.

**Comment by mikeblakeuk** [comment]
Hi @markcowl , thanks for the update.

- We have a dependency on a NuGet library that uses Storage 7.0.0. As per, the error at the top of this thread (`Unable to resolve dependencies.`)

- We want the latest, working, with fixes version of storage. I assume, as the nuget version is generated, support 6 AND 7 for just HDInsight.Jobs isn't "supported"?

- By 
> coordination  with other teams

Do you mean within Microsoft?(!) Why is storage not included in the SDK?
Why are there 3 NuGets, what one should we use? Why is Storage not in the *Azure* namespace?:
https://github.com/Azure/azure-storage-net
https://github.com/Azure/azure-sdk-for-net/tree/master/src/ServiceManagement/Storage and 
https://github.com/Azure/azure-sdk-for-net/tree/master/src/ResourceManagement/Storage

Why would we need to use PowerShell to upgrade?

Can we use Binding Redirect to overcome this issue?

NuGet change log: https://github.com/Azure/azure-storage-net/blob/master/changelog.txt
REST change log: https://msdn.microsoft.com/en-us/library/azure/dd894041.aspx
http://stackoverflow.com/questions/37173722/is-azure-storage-rest-api-version-2015-07-08-used-by-storage-client-v7-documen

**Comment by Sheliv** [comment]
Thank you, mark, I do not need to use this library, I will update.

Thanks again.
Sheli

**Comment by cormacpayne** [comment]
@mikeblakeuk it appears that this PR is obsolete, if there is no movement by the end of the week, we will close this PR

**Comment by mikeblakeuk** [comment]
No probs. I assume there is no drive to upgrade to the the latest Storage? Shame @markcowl didn't get back to me. BTW; I can't see why the build is failing as I don't have an account.

**Comment by markcowl** [comment]
@mikeblakeuk Unfortunately, binding redirects are only usable in applications.

As far as namespaces and repo membership are concerned, this is primarily a repo for management SDKs, and the storage team maintains their own repository for the storage data plane SDK.

Since it seems this change isn't needed by the feature team, closing for now.

**Pull request closed without merging**